### ISSUE TITLE
[Bug] partialBundling causes an error in the build product

### ISSUE THREAD

**Issue by niuiic** [farm-fe/farm#769]
### Steps to reproduce

```
git clone https://github.com/niuiic/gallery
pnpm i
pnpm build --filter css-components
pnpm dev --filter website
```

Open the console.

The output:
![image](https://github.com/farm-fe/farm/assets/63799565/2e60177f-72a6-45f1-a2e6-6f3f15626b30)

In `/home/niuiic/Documents/projects/gallery/apps/website/src/view/page/main/Main.tsx`
```typescript
import cssComponents from 'css-components'
console.log(cssComponents)
```
`cssComponents` is `{}`

In `/home/niuiic/Documents/projects/gallery/packages/css/farm.config.build.ts`
Unset `partialBundling`, then `cssComponents` is an array with one item.
`enforceResources` is set to `[{ name: 'wavyUnderline', test: [ 'src/components/wavyUnderline/*' ] }]`.
```typescript
import type { UserConfig } from '@farmfe/core'
import dts from '@farmfe/js-plugin-dts'
import { farmRawPlugin, farmSassPlugin } from 'farm-plugins'
import { readdirSync, statSync } from 'fs'
import { join } from 'path'

const chunks: {
  name: string
  test: string[]
}[] = []

const walkDir = (dirPath: string, cb: (filePath: string) => void) => {
  const items = readdirSync(dirPath)
  items.forEach((item) => {
    const itemPath = join(dirPath, item)
    const isDir = statSync(itemPath).isDirectory()
    if (isDir) {
      walkDir(itemPath, cb)
    } else {
      cb(itemPath)
    }
  })
}

const collectChunks = (filePath: string) => {
  if (!filePath.includes('index.ts')) {
    return
  }
  let matchedText = filePath.match(/\/src\/components\/([^\/]+)\/*/)
  if (matchedText) {
    chunks.push({
      name: matchedText[1],
      test: [`src/components/${matchedText[1]}/*`]
    })
  }
}

walkDir(join(process.cwd(), 'src/components'), collectChunks)

const config: UserConfig = {
  compilation: {
    input: {
      index: './src/index.ts'
    },
    output: {
      entryFilename: 'index.mjs',
      targetEnv: 'browser',
      format: 'esm'
    },
    resolve: {
      alias: {
        '@/': join(process.cwd(), 'src')
      }
    },
    minify: true,
    sourcemap: false,
    presetEnv: false,
    lazyCompilation: true
  },
  plugins: [
    farmSassPlugin(),
    farmRawPlugin(),
    dts({
      tsConfigPath: './tsconfig.json'
    })
  ]
}

export default config
```

### What is expected?

The output
![image](https://github.com/farm-fe/farm/assets/63799565/9baf5117-e153-4bfd-8875-951fda2466ec)

### What is actually happening?

`{}`

### System Info

```Shell
System:
    OS: Linux 6.1 Gentoo Linux
    CPU: (16) x64 AMD Ryzen 7 5800H with Radeon Graphics
    Memory: 12.92 GB / 15.59 GB
    Container: Yes
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 21.1.0 - /run/user/1000/fnm_multishells/11955_1700613217468/bin/node
    npm: 10.2.0 - /run/user/1000/fnm_multishells/11955_1700613217468/bin/npm
    pnpm: 8.10.5 - /run/user/1000/fnm_multishells/11955_1700613217468/bin/pnpm
  npmPackages:
    @farmfe/core: ^0.14.3 => 0.14.3
```
```


### Any additional comments?

_No response_

**Comment by niuiic** [comment]
In `/home/niuiic/Documents/projects/gallery/packages/css/src/index.ts`
```typescript
import * as wavyUnderline from './components/wavyUnderline'

interface Component {
  css: string
  html: string
}

export default [wavyUnderline] as Component[]
```

**Comment by wre232114** [comment]
use `partialBundling.groups` instead of `enforceResources`. `enforceResources` will ignore dynamic import contraints.

and `test` should be a regex, for example: `src/components/wavyUnderline/.*`

**Comment by niuiic** [comment]
works

**Comment by niuiic** [comment]
I made a mistake, it does not work. Setting groups does not split it into chunks.

set `enforceResources`
![image](https://github.com/farm-fe/farm/assets/63799565/f8433751-54af-42f1-9bd2-6377302324f4)

set `groups`
![image](https://github.com/farm-fe/farm/assets/63799565/61537cb8-7570-47a2-9a01-7ffa2e2d4bba)

And there is no dynamic import.

**Comment by wre232114** [comment]
try set `partialBundling.targetMinSize` to `0`

**Comment by niuiic** [comment]
Set groups and targetMinSize

```
farm-plugins:build: cache bypass, force executing 106922e0314b13a1
farm-plugins:build:
farm-plugins:build: > farm-plugins@0.0.0 build /home/niuiic/Documents/projects/gallery/packages/farm-plugins
farm-plugins:build: > farm build
farm-plugins:build:
farm-plugins:build: [ Farm ] Using config file at /home/niuiic/Documents/projects/gallery/packages/farm-plugins/farm.config.ts
farm-plugins:build: [ Farm ] ⚡️ Build completed in 32ms! Resources emitted to dist.
farm-plugins:build: [ FarmDtsPlugin ] ⚡️ Dts Plugin Build completed in 766ms! Resources emitted to dist.
farm-plugins:build: Failed to mark outputs as cached for farm-plugins#build: rpc error: code = Unavailable desc = cookie timeout expired
css-components:build: cache bypass, force executing 0697386bf6a25838
css-components:build:
css-components:build: > css-components@0.0.0 build /home/niuiic/Documents/projects/gallery/packages/css
css-components:build: > farm build -c farm.config.build.ts
css-components:build:
css-components:build: [ Farm ] Using config file at /home/niuiic/Documents/projects/gallery/packages/css/farm.config.build.ts
css-components:build: file:///home/niuiic/Documents/projects/gallery/node_modules/.pnpm/@farmfe+core@0.14.3/node_modules/@farmfe/core/dist/config/schema.js:264
css-components:build:         throw new Error(`${validationError}. \n Please check your configuration file or command line configuration.`);
css-components:build:               ^
css-components:build:
css-components:build: Error: Validation error: Number must be greater than 0 at "compilation.partialBundling.targetMinSize".
css-components:build:  Please check your configuration file or command line configuration.
css-components:build:     at parseUserConfig (file:///home/niuiic/Documents/projects/gallery/node_modules/.pnpm/@farmfe+core@0.14.3/node_modules/@farmfe/core/dist/config/schema.js:264:15)
css-components:build:     at resolveUserConfig (file:///home/niuiic/Documents/projects/gallery/node_modules/.pnpm/@farmfe+core@0.14.3/node_modules/@farmfe/core/dist/config/index.js:293:26)
css-components:build:     at async build (file:///home/niuiic/Documents/projects/gallery/node_modules/.pnpm/@farmfe+core@0.14.3/node_modules/@farmfe/core/dist/index.js:49:24)
css-components:build:
css-components:build: Node.js v21.1.0
css-components:build:  ELIFECYCLE  Command failed with exit code 1.
css-components:build: ERROR: command finished with error: command (/home/niuiic/Documents/projects/gallery/packages/css) pnpm run build exited (1)
css-components#build: command (/home/niuiic/Documents/projects/gallery/packages/css) pnpm run build exited (1)
```

set `enforceResources` and `targetMinSize`

```
farm-plugins:build: cache bypass, force executing 106922e0314b13a1
farm-plugins:build:
farm-plugins:build: > farm-plugins@0.0.0 build /home/niuiic/Documents/projects/gallery/packages/farm-plugins
farm-plugins:build: > farm build
farm-plugins:build:
farm-plugins:build: [ Farm ] Using config file at /home/niuiic/Documents/projects/gallery/packages/farm-plugins/farm.config.ts
farm-plugins:build: [ Farm ] ⚡️ Build completed in 34ms! Resources emitted to dist.
farm-plugins:build: [ FarmDtsPlugin ] ⚡️ Dts Plugin Build completed in 744ms! Resources emitted to dist.
farm-plugins:build: Failed to mark outputs as cached for farm-plugins#build: rpc error: code = Unavailable desc = cookie timeout expired
css-components:build: cache bypass, force executing 3f5023f3be6a3a68
css-components:build:
css-components:build: > css-components@0.0.0 build /home/niuiic/Documents/projects/gallery/packages/css
css-components:build: > farm build -c farm.config.build.ts
css-components:build:
css-components:build: [ Farm ] Using config file at /home/niuiic/Documents/projects/gallery/packages/css/farm.config.build.ts
css-components:build: file:///home/niuiic/Documents/projects/gallery/node_modules/.pnpm/@farmfe+core@0.14.3/node_modules/@farmfe/core/dist/config/schema.js:264
css-components:build:         throw new Error(`${validationError}. \n Please check your configuration file or command line configuration.`);
css-components:build:               ^
css-components:build:
css-components:build: Error: Validation error: Number must be greater than 0 at "compilation.partialBundling.targetMinSize".
css-components:build:  Please check your configuration file or command line configuration.
css-components:build:     at parseUserConfig (file:///home/niuiic/Documents/projects/gallery/node_modules/.pnpm/@farmfe+core@0.14.3/node_modules/@farmfe/core/dist/config/schema.js:264:15)
css-components:build:     at resolveUserConfig (file:///home/niuiic/Documents/projects/gallery/node_modules/.pnpm/@farmfe+core@0.14.3/node_modules/@farmfe/core/dist/config/index.js:293:26)
css-components:build:     at async build (file:///home/niuiic/Documents/projects/gallery/node_modules/.pnpm/@farmfe+core@0.14.3/node_modules/@farmfe/core/dist/index.js:49:24)
css-components:build:
css-components:build: Node.js v21.1.0
css-components:build:  ELIFECYCLE  Command failed with exit code 1.
css-components:build: ERROR: command finished with error: command (/home/niuiic/Documents/projects/gallery/packages/css) pnpm run build exited (1)
css-components#build: command (/home/niuiic/Documents/projects/gallery/packages/css) pnpm run build exited (1)

 Tasks:    1 successful, 2 total
Cached:    0 cached, 2 total
  Time:    3.367s
Failed:    css-components#build

 ERROR  run failed: command  exited (1)
 ELIFECYCLE  Command failed with exit code 1.
```

**Comment by wre232114** [comment]
my mistake, you can set `targetMinSize` to `1`, cause it can not be zero for now

**Comment by niuiic** [comment]
Three components and three output files. This works. But it seems there is still something wrong with `enforceResources`.

![image](https://github.com/farm-fe/farm/assets/63799565/9308c322-6dc5-4c51-b397-d52a2c0e2d95)

Test should be correct now.

```
css-components:build: [
css-components:build:   {
css-components:build:     name: 'wavyUnderline',
css-components:build:     test: [ 'src/components/wavyUnderline/.*' ]
css-components:build:   }
css-components:build: ]
```

**Comment by wre232114** [comment]
I'll check `enforceResources`, it should work too.

**Comment by niuiic** [comment]
![image](https://github.com/farm-fe/farm/assets/63799565/f5cd1834-1bed-4faf-9549-08eaf78e3cdc)

`wavyUnderline/index.ts` import raw string from `wavyUnderline/wavyUnderline.html` and `wavyUnderline/wavyUnderline.scss`, and then export these.

`index.ts` export all thing from `wavyUnderline/index.ts`.

**Comment by niuiic** [comment]
Unfortunately, `groups` does not work either.

I forget to test the website.

![image](https://github.com/farm-fe/farm/assets/63799565/bc6755ed-9de7-48e0-80b6-a59f0957c666)

**Comment by wre232114** [comment]
This may be a issue related to bundling, I'll look into it

**Comment by wre232114** [comment]
I found the reason, it will be fixed in `0.14.4`

**Comment by niuiic** [comment]
This has not been fixed in `0.14.4`.
![image](https://github.com/farm-fe/farm/assets/63799565/6b81c67a-5763-4030-84e0-fed233143a14)
@wre232114

**Comment by wre232114** [comment]
I'll try, it works with my examples

**Comment by wre232114** [comment]
`entryFilename: 'index.mjs'` should be `entryFilename: '[entryName].mjs'`. `index.mjs` will overide all entries name including the farm runtime.

I'll make farm runtime not affect by this option in 0.14.5 too.

**Comment by wre232114** [comment]
fixed in `0.14.5`

**Comment by niuiic** [comment]
This is fixed.
